app/services/foreman/plugin.rb

```diff
@@ -207,7 +208,8 @@ def permission(name, hash, options = {})
 
     # Add a new role if it doesn't exist
     def role(name, permissions)
-      return false if pending_migrations
+      @default_roles[name] = permissions
+      return false if pending_migrations || Rails.env.test?
```

Comment:
FYI this is the change I made, added `Rails.env.test?`.

To give you the full context @domcleal @eLobato during tests we do not seeded data, we have fixtures and factorygirl. Therefore there is no point in creating default roles there. Actually it blocks Discovery from modifying existing role "Discovery Reader/Manager" from running tests:

```
/home/lzap/work/foreman/app/models/role.rb:131:in `add_permissions': some permissions were not found (ArgumentError)
        from /home/lzap/work/foreman/app/models/role.rb:146:in `add_permissions!'
```

I'd appreciate if you could take a quick look and merge this. Tests are green.